test/ao/sparsity/test_pruner.py

```diff
@@ -155,24 +140,14 @@ def test_constructor_cuda(self):
         assert 'test' in pruner.module_groups[0]
         assert pruner.module_groups[0]['test'] == 3
 
-    def test_prepare_linear(self):
-        model = Linear()
-        x = torch.ones(128, 16)
-        pruner = SimplePruner(model, None, None)
-        pruner.prepare()
-        for g in pruner.module_groups:
-            module = g['module']
-            # Check mask exists
-            assert hasattr(module, 'mask')
-            # Check parametrization exists and is correct
-            assert parametrize.is_parametrized(module)
-            assert hasattr(module, "parametrizations")
-            # Assume that this is the 1st/only parametrization
-            assert type(module.parametrizations.weight[0]) == PruningParametrization
-        assert model(x).shape == (128, 16)
+    def test_constructor(self):
+        self._test_constructor_on_device(torch.device('cpu'))
+        if torch.cuda.is_available():
+            self._test_constructor_on_device(torch.device('cuda'))
```

Comment:
Can we use https://github.com/pytorch/pytorch/blob/master/test/quantization/core/test_workflow_module.py#L674 directly?

Reply:
yeah, also probably just loop through a list of devices and don't use hypothesis.